ofctrl/fgraphFlow.go

```diff
@@ -248,8 +250,12 @@ func (self *Flow) xlateMatch() openflow13.Match {
 	}
 
 	// Handle Vlan id
-	if self.Match.VlanId != 0 {
-		vidField := openflow13.NewVlanIdField(self.Match.VlanId, nil)
+	if self.Match.NonVlan {
+		vidField := openflow13.NewVlanIdField(0, nil)
+		vidField.Value = new(openflow13.VlanIdField)
```

Comment:
What about `vidField.Value = openflow13.VlanIdField{}`, and then you don't transform from pointer to object again when setting ofMatch.AddField?

Reply:
There is no different between `openflow13.VlanIdField{}` and `new(openflow13.VlanIdField)`.
And `ofMatch.AddField(*vidField)` is still required even we change this line.

Reply:
PR comments updated.